cmd/centre/main.go

```diff
@@ -100,6 +100,40 @@ func lookupIP(hostFile string, addr string) ([]net.IP, error) {
 	return net.LookupIP(addr)
 }
 
+func lookupHostname(hostFile string, ip net.IP) (string, error) {
+	var err error
+	// First try the override
+	if hostFile != `` {
+		// Read the file and walk each line looking for a match.
+		// We do this so you can update the file without restarting the server
+		var f *os.File
+		if f, err = os.Open(hostFile); err != nil {
+			return "", err
+		}
+		// We're going to be real simple-minded. We take each line to consist of
+		// one IP, followed by one or more whitespace-separated hostnames
+		scan := bufio.NewScanner(f)
+		for scan.Scan() {
+			fields := strings.Fields(scan.Text())
+			// You have to have at least 3 fields: <ip> <hostname>... <mac addr>
```

Comment:
This doesn't seem to handle comments. Take a look at the standard library: https://github.com/golang/go/blob/3a3b8164fdcb071955284c13cda6ee0f29fc8bd3/src/net/hosts.go#L72